File: leetcode/javascript/1-two-sum.js
/**
 * https://leetcode.com/problems/two-sum/
 */

/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
 var twoSum = function(nums, target) {
  const seen = {};
  
  for (let i = 0; i < nums.length; i++) {
    const remainder = target - nums[i];
    
    if (seen[remainder] !== undefined) {
      return [seen[remainder], i];
    }
    
    seen[nums[i]] = i; 
  }
};

File: leetcode/go/1-two-sum.go
/**
 * https://leetcode.com/problems/two-sum/
 */

package main

func twoSum(nums []int, target int) []int {
	seen := make(map[int]int)

	for i, num := range nums {
			remainder := target - num

			if j, ok := seen[remainder]; ok {
					return []int{j, i}
			}

			seen[num] = i
	}

	return []int{}
}
